pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -491,82 +504,96 @@ func loadIPSecKeys(r io.Reader) (int, uint8, error) {
 	scanner.Split(bufio.ScanLines)
 	for scanner.Scan() {
 		var oldSpi uint8
-		var authkey []byte
-		offset := 0
+		var aeadKey, authKey []byte
+		offsetBase := 0
 
 		ipSecKey := &ipSecKey{
 			ReqID: 1,
 		}
 
-		// Scanning IPsec keys formatted as follows,
-		//    auth-algo auth-key enc-algo enc-key
+		// Scanning IPsec keys with one of the following formats:
+		// 1. [spi] aead-algo aead-key icv-len
+		// 2. [spi] auth-algo auth-key enc-algo enc-key [IP]
 		s := strings.Split(scanner.Text(), " ")
-		if len(s) < 2 {
-			return 0, 0, fmt.Errorf("missing IPSec keys or invalid format")
+		if len(s) < 3 {
```

Comment:
Considering adding a constant for `3` as well next to the offsets so it's easy to see the offsets don't go past the last entry.

Reply:
That's not the last entry. I considered making it a constant but I don't think it would improve readability. The format is documented 2 lines above and we can see that the secret should have at the minimum 3 fields. I've added a comment instead.